Jint/Runtime/Interop/DefaultTypeConverter.cs

```diff
@@ -41,182 +41,265 @@ public virtual object Convert(object value, Type type, IFormatProvider formatPro
+                    typeCode == TypeCode.UInt32 ||
+                    typeCode == TypeCode.UInt64)
+                {
+                    return (Engine engine, ITypeConverter typeConverter, object value, IFormatProvider f) => Enum.ToObject(type, value);
+                }
+                if (typeCode == TypeCode.Single)
+                {
+                    return (Engine engine, ITypeConverter typeConverter, object value, IFormatProvider f) => Enum.ToObject(type, (int)(float)value);
+                }
+                if (typeCode == TypeCode.Double)
                 {
-                    throw new ArgumentOutOfRangeException();
+                    return (Engine engine, ITypeConverter typeConverter, object value, IFormatProvider f) => Enum.ToObject(type, (int)(double)value);
                 }
+                else
+                {                    
+                    return (Engine engine, ITypeConverter typeConverter, object value, IFormatProvider f) =>
+                    {
+                        var integer = System.Convert.ChangeType(value, typeof(int), formatProvider);
+                        if (integer == null)
+                        {
+                            throw new ArgumentOutOfRangeException();
+                        }
 
-                return Enum.ToObject(type, integer);
+                        return Enum.ToObject(type, integer);
+                    };
+                }                    
             }
 
-            var valueType = value.GetType();
             // is the javascript value an ICallable instance ?
             if (valueType == typeof(Func<JsValue, JsValue[], JsValue>))
             {
-                var function = (Func<JsValue, JsValue[], JsValue>)value;
-
                 if (type.IsGenericType)
                 {
                     var genericType = type.GetGenericTypeDefinition();
 
                     // create the requested Delegate
                     if (genericType.Name.StartsWith("Action"))
                     {
-                        var genericArguments = type.GetGenericArguments();
-
-                        var @params = new ParameterExpression[genericArguments.Count()];
-                        for (var i = 0; i < @params.Count(); i++)
-                        {
-                            @params[i] = Expression.Parameter(genericArguments[i], genericArguments[i].Name + i);
-                        }
-                        var tmpVars = new Expression[@params.Length];
-                        for (var i = 0; i < @params.Count(); i++)
+                        return (Engine engine, ITypeConverter typeConverter, object value, IFormatProvider f) =>
                         {
-                            var param = @params[i];
-                            if (param.Type.IsValueType)
+                            var genericArguments = type.GetGenericArguments();
+
+                            var @params = new ParameterExpression[genericArguments.Count()];
```

Comment:
I realize it's not part of your changes... feel free to leave it if you feel you have done enough.